Add cmd-test-coreos-installer

cgwalters opened the pull request:
We have no coverage of coreos-installer today.  This was a quick
hacky shell script I wrote a few months ago and posted as a gist:
https://gist.github.com/cgwalters/bf6f5b6f788d01211dbe6cd362309a0d

This could obviously be cleaned up and de-duped with kola and
other cosa bits, but for now, let's just get some coverage.

Specifically e.g. one thing I want to do is support running
a specific kola suite (maybe just `basic`) against the newly installed image.

cgwalters commented on src/cmd-test-coreos-installer:
```diff
@@ -0,0 +1,120 @@
+
+completionf=${statedir}/completion.txt
+touch "${completionf}"
+qemu-system-x86_64 -accel kvm -m 8192 -nographic \
```
This wants https://github.com/coreos/mantle/pull/1147

cgwalters replied:
Will do this later

cgwalters commented on src/cmd-test-coreos-installer:
```diff
@@ -0,0 +1,120 @@
+# boots and writes a success message to a virtio-serial port, which we read on the host.
+set -euo pipefail
+
+run_tmp_webserver() {
```
Ohh and I forgot about https://github.com/coreos/mantle/pull/1141 will change to use that

darkmuggle commented on src/cmd-test-coreos-installer:
```diff
@@ -0,0 +1,120 @@
+
+completionf=${statedir}/completion.txt
+touch "${completionf}"
+qemu-system-x86_64 -accel kvm -m 8192 -nographic \
```
Greedy :) 

This might be a problem if we want to plumb it into the pipelines.

cgwalters commented:
Updated :new: 

 - Dropped RAM requirement
 - Uses `kola http-server`
 - Uses the ignition hack-3to2 to support RHCOS too

(I do want to rewrite this in Go in kola sometime soon but I don't think it's worth waiting for that)

jlebon commented on src/cmd-test-coreos-installer:
```diff
@@ -0,0 +1,101 @@
+buildid=$(jq -er .buildid < ${buildmeta})
+metalimg=$(jq -er .images.metal.path < ${buildmeta})
+instkernel=$(jq -er .images.kernel.path < ${buildmeta})
+instinitramfs=$(jq -er .images.initramfs.path < ${buildmeta})
```
In light of https://github.com/coreos/fedora-coreos-tracker/issues/203, we'll want to at least for now make this configurable to use the live versions if possible. The kargs interface should be the same otherwise so we shouldn't have to change anything else.

We can do this in a follow-up though if you prefer!

cgwalters replied:
Yeah I just want to get *any* coverage of the installer path for RHCOS right now.

jlebon commented on src/cmd-test-coreos-installer:
```diff
@@ -0,0 +1,101 @@
+cd -
+
+case "${metalimg}" in
+  *.raw) echo "NOTICE: Metal image is not compressed, doing so now temporarily"
```
Ahh, the new installer supports uncompressed payloads as well. But yeah, legacy one doesn't. Anyway, worth starting with the more common scenario.

jlebon commented on src/cmd-test-coreos-installer:
```diff
@@ -0,0 +1,101 @@
+  exit 1
+fi
+echo "ok metal+installer for ${buildid}"
+exit 0
```
Spurious?
